### PR TITLE
Improve toc

### PR DESCRIPTION
**Pull request by dafelix42**
- Detection of left side number
- Update regex for roman number to detect them on the right
- Condition of ratio integer/number detected 
- Condition of ratio ascending order integer/integer    (to improve, sometimes detect numbers that are not related of TOC and give wrong condition)

followings, 
screenshot of toc detection with this branch
screenshot of toc detection with develop branch

axa_half_year
axa_half_year_develop
backgroundLine
backgroundLine_develop
backgroundLine2
backgroundLine2_develop
ccesAnnualreport
ccesAnualReport_develop
curaUserManualV1 0
curaUserManualV1 0_develop
dankseBank2019
dankseBank2019_1_develop
dankseBank2019_2_develop
iosSecurityGuide
iosSecurityguide_develop
toc001_20200519
toc001_20200519_develop
toc002_20200519
toc002_20200519_develop
toc003
toc003_develop
toc006
toc006_develop
toc015
toc015_develop
ultimaker_2_go
ultimaker_2_go_develop
ultimakerCura
ultimakerCura_develop
userManual_UM2
userManual_UM2_develop

### CODE REVIEW

File: server/src/processing/TableOfContentsDetectionModule/TableOfContentsDetectionModule.ts [hunk 1]
```diff
@@ -34,11 +35,19 @@ export class TableOfContentsDetectionModule extends Module<Options> {
   }
 
   public main(doc: Document): Document {
```
Comment (jvalls-axa) [inline server/src/processing/TableOfContentsDetectionModule/TableOfContentsDetectionModule.ts:37]: @dafelix42  

135 lines in main function is not best way to code....

Could you please try to split main into small functions in order to make code more human understandable??

File: server/src/processing/TableOfContentsDetectionModule/detection-methods.ts [hunk 2]
```diff
@@ -26,29 +27,53 @@ const detectionMethods = {
...
+      .getWords()
+      .filter(
+        word => BoundingBox.getOverlap(word.box, intersectionBoxLeft).box1OverlapProportion > 0,
+      )
+      .filter(word => !isSeparator(word));
+    // logger.info('word right= ' + wordsInsideIntersectionRight.toString());
+    // logger.info('wrd r= ' + wordsInsideIntersectionRight.filter(isNumberRight));
+    // logger.info('word left= ' + wordsInsideIntersectionLeft.toString());
+    // logger.info('wrd l= ' + wordsInsideIntersectionLeft.filter(isNumberLeft));
+    // logger.info('word left= ' + wordsInsideIntersectionLeft.toString());
```
Comment (jvalls-axa) [inline server/src/processing/TableOfContentsDetectionModule/detection-methods.ts:50]: Remove unused code please

File: server/src/processing/TableOfContentsDetectionModule/TableOfContentsDetectionModule.ts [hunk 3]
```diff
@@ -47,8 +56,100 @@ export class TableOfContentsDetectionModule extends Module<Options> {
...
+        }
+        nbNumber = allStoredNumbers.length;
+        nbInteger = allStoredInteger.length;
+
+        const allIntegerParam = [];
+
+        // Find for each stored integer in array how many of integer in the followings indexes are >=
+        // then it is stored in an array integerParam [integer, integerSup, indexOfarray]
+        if (nbInteger > 2) {
+          let j = 0;
```
Comment (jvalls-axa) [inline server/src/processing/TableOfContentsDetectionModule/TableOfContentsDetectionModule.ts:94]: Try to use param names that makes it easy to understand

File: package.json [hunk 4]
```diff
@@ -38,7 +38,7 @@
   "husky": {
     "hooks": {
       "pre-commit": "lint-staged",
-      "pre-push": "npm test && npm audit && npm run remark"
+      "pre-push": "npm test"
```
Comment (jvalls-axa) [inline package.json:41]: why ?

Comment (dafelix42) [reply package.json]: I have put back the initial version. In the past I had some trouble to push with npm audit  and run remark. I didn't want to push this modfication. Now It works fine.

**Comment by dafelix42** [comment]
With this new commits comes:
- Detection of aligned number/roman number and check of ascending order within each group of aligned number
- Detect entire line of TOC even if numbers are in different paragraphs than text
- Condition of 2 numbers in ascending order minimum to be a TOC 

Followings,
screenshot of toc detection with this branch
screenshot of toc detection with develop branch

sampleTOC001
sampleTOC001_develop
sampleTOC003
sampleTOC003_develop
sampleTOC008
sampleTOC008_develop
ultimaker2goTOC
ultimaker2goTOC_develop
sampleTOC006
sampleTOC006_develop

Some cases that currently doesn't work:
- TOC with roman number not linked with dots
sampleTOC002b
- Many paragraphs on same page as TOC
cces_anual_reportTOC
- TOC with roman number on the left side is not implemented (most of time it's to numerate titles/chapters)
sampleTOC007

**Comment by dafelix42** [comment]
Solve this previous case that wasn't working:
TOC with roman number not linked with dots

Some exemples:
Screenshot 2020-06-17 at 18 56 51
Screenshot 2020-06-17 at 18 58 15
Screenshot 2020-06-17 at 18 57 42
Screenshot 2020-06-17 at 18 59 01
Screenshot 2020-06-17 at 18 59 46

**Pull request merged**